ops/pebble.py

```diff
@@ -752,5 +861,198 @@ def get_services(self, names: List[str] = None) -> List[ServiceInfo]:
         query = None
         if names is not None:
             query = {'names': ','.join(names)}
-        result = self._request('GET', '/v1/services', query)
-        return [ServiceInfo.from_dict(info) for info in result['result']]
+        resp = self._request('GET', '/v1/services', query)
+        return [ServiceInfo.from_dict(info) for info in resp['result']]
+
+    def read_file(self, path: str, destination: BinaryIO):
```

Comment:
It feels odd that this is
`read_file(path: str, destination: BinaryIO)`
rather than
`read_file(path: str) => BinaryIO`
I realize this somewhat exists to support bulk requests, but it does feel awkward.

I also wonder about a convenience function:
`def read_content(self, path: str) => bytes?/str?`

But that could be written on top of any sort of file based api.

Reply:
Per discussion with Ian on Mattermost, it's actually a bit tricky to have this return a readable file-like object, because you're not yet finished reading the multipart response by the time `read_file` returns. So it makes the implementation tricky (you'd have to return some kind of object which reads bytes and then goes on to parse the metadata JSON in the last `read` call), and it makes error handling a bit weird, because protocol/path errors may be raised by the returned object's `read()` rather than the main `read_file()` call.

So what I've decided is to just include `read_content` / `write_content` for now, as those are the simplest APIs and we need those APIs anyway (and my implementation was reading everything into memory anyway). This will get us going, and we can always add streaming/bulk/whatever other APIs in a future release.